--- a/playbook/roles/kubelet-config/templates/kubelet.service.j2
+++ b/playbook/roles/kubelet-config/templates/kubelet.service.j2
@@ -20,7 +20,7 @@ ExecStart=/opt/kubernetes/hyperkube kubelet \
   --pod-infra-container-image={{ k8s_pause_image }} \
   {% if cloud_provider != 'none' -%}
   --cloud-provider={{ cloud_provider }} \
-  {% if inventory_hostname in groups['master'] -%}
+  {% if ((inventory_hostname in groups['master']) or (cloud_provider == "azure")) -%}
   --cloud-config={{ cloud_provider_conf }} \
   {% endif -%}
   {% endif -%}
